### PR TITLE
Adds combat tentacles for changelings

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Changelings can now buy Tentacles on the Cellular emporium for 2 evolution points.
add: Tentacles, once used, can be fired once against an item or mob to pull it towards yourself. Items will be automatically grabbed. Costs 10 chemicals per tentacle.
add: On humanoid mobs tentacles have a varying effect depending on intent:
- Help intent simply pulls the target closer without harming him;
- Disarm intent does not pull the target but instead pulls whatever item he's holding in his hands to yours;
- Grab intent puts the target into an aggressive grab after it is pulled, allowing you to throw it or try to consume it;
- Harm intent will briefly stun the target on landing; if you're holding a sharp weapon you'll also impale the target, dealing increased damage and a longer stun.
/:cl:

Some shots of tentacles in action: 
https://i.gyazo.com/6a25b79a9d96fc2a05dd8cc33095c7a0.gif
https://i.gyazo.com/f13e9510678f3bc29b7fdc3ca3fedb0f.gif
https://i.gyazo.com/4ce745d4543b511b4e593760517bf57b.gif

Ignore the armblade, in this PR the tentacle is not visible until fired.
Speaking of sprites, these are kinda horrible and could use a real spriter's work if someone is available.

Reason for the PR: this seems like a cool tool to give to changelings, which gives them another option other than armblade for fighting or utility.

Collateral changes: Beam() can now customize the update rate of the beam.

Note for commenters: hentai jokes have already been done in coderbus. Don't get repetitive.

### CODE REVIEW

**Comment by Razharas** [comment]
Extreme ERP potential 👌 👍

**Comment by ExcessiveUseOfCobblestone** [comment]
xXMLG Ling Hookshot MontageXx

This IS a "skillshot" and not a one click tracer, right?

**Comment by XDTM** [comment]
skillshot indeed

**Comment by PKPenguin321** [comment]
Might be cool as a reach weapon if those get added but yeah this is super fucking cool :+1:

File: code/game/gamemodes/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -191,6 +194,160 @@
...
+		return 0
+	var/mob/living/carbon/human/H = firer
+	if(istype(target, /obj/item))
+		var/obj/item/I = target
+		if(!I.anchored)
+			firer << "<span class='notice'>You pull [I] towards yourself.</span>"
+			H.throw_mode_on()
+			I.throw_at(H, 10, 2)
+			. = 1
+			spawn(10)
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/mutations.dm]: spawn is banned.

**Comment by XDTM** [comment]
Code has been despawned

**Comment by dannno** [comment]
coooooooooool

File: code/game/gamemodes/changeling/powers/mutations.dm [hunk 2]
```diff
@@ -191,6 +194,167 @@
...
+	user << "<span class='warning'>The [name] is not ready yet.<span>"
+
+/obj/item/ammo_casing/magic/tentacle
+	name = "tentacle"
+	desc = "a tentacle."
+	projectile_type = /obj/item/projectile/tentacle
+	caliber = "tentacle"
+	icon_state = "tentacle_end"
+	firing_effect_type = null
+	var/obj/item/weapon/gun/magic/tentacle/gun //the item that shot it
```
Comment (optimumtact) [inline code/game/gamemodes/changeling/powers/mutations.dm]: make a custom destroy for this that nulls this

File: code/game/gamemodes/changeling/powers/mutations.dm [hunk 3]
```diff
@@ -191,6 +194,167 @@
...
+
+/obj/item/projectile/tentacle
+	name = "tentacle"
+	icon_state = "tentacle_end"
+	pass_flags = PASSTABLE
+	damage = 0
+	damage_type = BRUTE
+	range = 8
+	hitsound = 'sound/weapons/thudswoosh.ogg'
+	var/chain
```
Comment (optimumtact) [inline code/game/gamemodes/changeling/powers/mutations.dm]: custom destroy nulling your custom vars as well thanks

**Comment by Armhulen** [comment]
new sprites, then have a tentacle arm replace the dark blessing (though no pulling, just tentacle arm)

**Comment by kilkun** [comment]
does ling really need more buffs?

they already have free thermals, free healing, free adrenals, free anti-cuff, free revival, free voice changer, free space suit and free armor, not to mention at least 3 different "get out of jail free" cards, one of which kills an unarmored person in close proximity.

is giving them an ability to literally take guns out of people's hands and stun them a good idea? I mean dont get me wrong, it's cool and all but for fucks sake tone down some other of their abilities or something. they are becoming wizard-tier powerful.

I say make it cost 6 points. that way it severely limits what other abilities they can take. its also super versatile so its not like it isn't worth the 6 points anyways.

**Comment by kilkun** [comment]
at this rate the next change to changeling will be energy shield mutations.

**Comment by PKPenguin321** [comment]
note to future us after this is merged: maybe make the disarm one not stun

**Comment by kilkun** [comment]
"""""maybe"""""

for what little my shitty opinions matter, i say dont merge it until all the stuns are removed. changelings not being able to stun is the only thing between them and being more broken than shadowlings.

**Comment by PKPenguin321** [comment]
what's the point of the pull if it doesn't stun though

**Comment by kilkun** [comment]
cryo sting to slow them down.

**Comment by kilkun** [comment]
follow it up with beating them over the head with your armblade while they fail to yakkety sax

**Comment by kilkun** [comment]
also you could pull them over a slip for a stun and i can think of a few other ways to weaponize it but im not posting them on a public forum. get creative and remember all the proximity shit that exists in the game.

**Comment by kilkun** [comment]
also i maintain 2 points is really REALLY **REALLY** cheap :australia:

**Comment by AdamElTablawy** [comment]
hence why i am still working on scrapping ling and replacing it with goonling

**Comment by kilkun** [comment]
looking forward to that

especially the part where people reject it because it no longer means godmode.

**Comment by dannno** [comment]
add a tentacle as a chaplain null rod skin
like east of west
i dont give a fuck how broken it is

**Comment by kilkun** [comment]
honestly everybody should just get tentacles for free.

its the only way to balance this. :colbert:

**Comment by AdamElTablawy** [comment]
@kilkun 
if anything, goonstation changeling is MORE op than tgling, they just have less potency early game in 1v1

**Comment by kilkun** [comment]
that is the whole reason why tgling sucks. its too fucking easy. you can evolve all your powers from the start and dont have to work for anything. all eating people does is let you readapt your powers but why do that when there's one loadout that gives you everything you need and then supplement everything else by looting people you murderbone.

there's a lot that can go wrong between roundstart and horror form with lljkling.

**Comment by optimumtact** [comment]
goon players go home

**Comment by kilkun** [comment]
goon is objectively the best codebase, ignoring the fact that antag roles dont hold your hand, there's the whole stun system which is leagues ahead of anything else.

if the community would stop bickering for 5 seconds and take the time to learn how to play goon everybody would see that ss13 doesnt have to have god awful combat entrenched in garbo MUD pitfalls and that it can actually be fun... even _with_ stuns!

a shame that you have to choose between noRP nerdfest and medRP extended rounds.

**Comment by optimumtact** [comment]
Well then you won't mind if you are permabanned here so you can go play on goon then right?

**Comment by dannno** [comment]
haha what does that even mean oranges

**Comment by kilkun** [comment]
i think that last sentence explains exactly why i would mind, but permabanning for telling you that our garbage stuns are garbage is a little more than mildly petty.

**Comment by AdamElTablawy** [comment]
@optimumtact 

go play on goon so you can like it because it has good things and bad things

what /tg/ lacks goon has and vice versa and i have never seen a legitimately well thought out complaint about it

**Comment by optimumtact** [comment]
I can't, they permabanned me 5 years ago for reading lusty xeno maid and I've held the grudge ever since.

**Comment by XDTM** [comment]
By the way the disarm one does not stun anymore, it was just a bug in testing. The only way to stun with this is to use the harm intent, and it stuns for 1-2 seconds or 3-4 if you have a sharp weapon; or alternatively you can grab and table to get a longer but less guaranteed stun.

Speaking of balance, i agree that changelings having all their powers from the start is kinda bad. I was planning to rework changelings into an evolving antag/side-antag that could get skills/passives for different playstyles.

**Comment by KorPhaeron** [comment]
Very cool. Could use a new arm sprite though.

**Comment by dannno** [comment]
tomorrow i might try to make one but no guarantees, im retardded

**Comment by kevinz000** [comment]
goon lings suck.
@XDTM I think it's kinda overpowered to give changelings a (spammable?) ranged ability with the power to disarm/stun.

**Comment by XDTM** [comment]
10 chemicals a shot, no damage, and the stun is fairly short. Also you can dodge it.

**Comment by Grazyn** [comment]
Judging from the gif the armblade counts as "sharp weapon" so harm tentacle effectively turn into one-hit kill when they get stunned, impaled, and you follow up with a couple more armblade hits. Seems pretty OP to me

**Comment by jughu** [comment]
Then don't get caught by the tentacle

**Comment by jughu** [comment]
If the ling has its armblade out and yoy get tentacled by it which is basically a weaker taser you deserve it tbh

**Comment by XDTM** [comment]
I could reduce the multiplier, i guess. Double damage could end up being op.

**Comment by Shapsy** [comment]
But I don't understand why lings need an inherent taser power to begin with. Was using your arm blade for chasing people who had a slight chance of getting away if they were lucky not fun enough for the ling, so they have to get an innate ranged weapon too?

**Comment by jughu** [comment]
If you add this we might aswell remove adrenals

**Comment by XDTM** [comment]
Which wouldn't be a bad idea, really.
To be really honest my reason for this pr is that i really wanted to give lings more body horror abilities instead of several different invisible healing/antirestraint abilities.
Also, i'm planning a ling rework, and since i coded this and it works by itself, i wanted some feedback on it.

**Comment by jughu** [comment]
I wouldnt mind changelings be strong tbh they shouldnt just have abilities like adrenals with this which would make them technically invincible get tased adrenal tentacle take their taser shoot them

**Comment by XDTM** [comment]
Disarm can be a risk since it does nothing at all if the target has nothing in hand. Plus, since it throws the item to your position and not further, you won't catch it if you're yakety saxing too much. 
Also there is a possibility that you could stun yourself by grabbing stunbatons, but i'm not sure how grab code behaves there.

**Comment by AdamElTablawy** [comment]
but that's classic literature

**Comment by PKPenguin321** [comment]
Mute sting is absolutely not a staple. It actually kind of sucks imo

**Comment by Armhulen** [comment]
@dannno please make this a null rod item

**Comment by dannno** [comment]
o-ok?

**Comment by FantasticFwoosh** [comment]
Swap out armblade directly for this and it'll be 100% better trust me for actually keeping ling a stealth/utility antag rather than ('look how many people i can stab ma, while i vent in plasma and abuse my regeneration')

**Comment by kilkun** [comment]
mute sting would be better if you could whisper while muted.

i agree though, either make this cost a shit ton more or take out one of their other high profile abilities esp. adrenals. this seriously pushes ling into hardcore easymode territory.

**Comment by dannno** [comment]
maybe just make it so you can either take armblade or tentacle, not both
idk

**Comment by kilkun** [comment]
armblade isnt that overpowered though. to use it in conjunction with this requires coordination beyond your average hubbie who are now the majority server population, at least on sybil.

i think making it cost 6 points is fair. you get a lot of mileage with 4 individual uses.

**Comment by jughu** [comment]
People saying having a armblade would be OP with tentacles
But if you get tentacled by a person who extends or already have his armblade out you kinda deserve it tbh

**Comment by Alek2ander** [comment]
If it hits a wall or doesn't hit anything at all it should pull _you_ towards the targeted point.

**Comment by XDTM** [comment]
@kilkun Just to clear, an armblade is an undectectable, undisarmable, permanent, almost esword tier weapon that requires no storage space, is sharp and can force open doors, and it costs two points.
Is a one-use 2-4 second stun really worth three times as much?

**Comment by DorfGuy1** [comment]
hello, anybody else think that using the hold intent on someones crotch with this should do horrible things, giving target brain damage.

**Comment by kilkun** [comment]
yes because:
a.) not one use, its 7 uses.
b.) you literally can steal guns with it

changelings are way too overpowered right now, they don't need more really good abilities.

**Comment by optimumtact** [comment]
Feel free to make a salty revert pr, but only after at least 5 days.

**Pull request merged**